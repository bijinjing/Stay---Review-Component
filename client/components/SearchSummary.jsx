import React from 'react'
import styled from 'styled-components';

const Search_summary = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top:10px;
  padding-left:40px;
  padding-bottom:10px;
  border-bottom: 1px ridge lightgrey
`;
const Back_to_all_button = styled.a`
  font: inherit
  text-align: right
  color: #008489;
  cursor: pointer
`;

class SearchSummary extends React.Component{
  constructor(props) {
    super(props)
    this.backToAllHandler = this.backToAllHandler.bind(this)
  }

  backToAllHandler(){
    this.props.clearSearchValue()
  }


  render(){
    return(
        <Search_summary>
          {this.props.searchCount > 1?
            <span>{this.props.searchCount} guests have mentioned "<a style={{fontWeight:`bold`}}>{this.props.searchValue}</a>"</span> :
              this.props.searchCount == 1?
                <span>1 guest has mentioned "<a style={{fontWeight:`bold`}}>{this.props.searchValue}</a>"</span>:
                <span>None of guests have mentioned "<a style={{fontWeight:`bold`}}>{this.props.searchValue}</a>"</span>
          }
          <Back_to_all_button id="back_to_all" onClick={this.backToAllHandler}>Back to all reviews</Back_to_all_button>
        </Search_summary>
    )
  }
}

export default SearchSummary