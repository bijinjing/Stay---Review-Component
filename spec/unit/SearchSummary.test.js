import React from 'react';
import SearchSummary from '../../client/components/SearchSummary.jsx';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import {mount} from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('SearchSummary', ()=>{
  it('renders without crashing when there is a search result', () => {
    const props = {
      searchCount:5,
      searchValue:"hello"
    }

    const wrapper = shallow(<SearchSummary {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders without crashing when there is no search result', () => {
    const props = {
      searchCount:0,
      searchValue:"hello"
    }

    const wrapper = shallow(<SearchSummary {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should run backToAllHandler with backToAll click', () => {
    let result = null;
    const props = {
      clearSearchValue:()=>{result="hit"}
    }
    const component = mount(<SearchSummary {...props}/>);
    component
      .find('#back_to_all')
      .simulate('click',);

    expect(result).toEqual("hit");
    component.unmount();
  });
})