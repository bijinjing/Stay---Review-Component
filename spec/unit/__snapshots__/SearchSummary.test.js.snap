// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSummary renders without crashing when there is a search result 1`] = `
<div>
  <a>
    5
     guests have mentioned "
    hello
    "
  </a>
  <a
    id="back_to_all"
    onClick={[Function]}
  >
    Back to all reviews
  </a>
</div>
`;

exports[`SearchSummary renders without crashing when there is no search result 1`] = `
<div>
  <a>
    None of guests have mentioned "
    hello
    "
  </a>
  <a
    id="back_to_all"
    onClick={[Function]}
  >
    Back to all reviews
  </a>
</div>
`;
